File: src/messageprocessor/src/main/java/currencyfair/messageprocessor/Trade.java
package currencyfair.messageprocessor;

import java.util.Date;

public class Trade {
	private long tradeId;
	private String userId;
	private String currencyFrom;
	private String currencyTo;
	private double amountSell;
	private double amountBuy;
	private double rate;
	private Date timePlaced;
	private String originatingCountry;
	private Date timeReceived;
	private int processingStatus;
	private Date timeProcessed;
	private boolean exception;

	public String getUserId() {
		return userId;
	}
	public void setUserId(String userId) {
		this.userId = userId;
	}
	public String getCurrencyFrom() {
		return currencyFrom;
	}
	public void setCurrencyFrom(String currencyFrom) {
		this.currencyFrom = currencyFrom;
	}
	public String getCurrencyTo() {
		return currencyTo;
	}
	public void setCurrencyTo(String currencyTo) {
		this.currencyTo = currencyTo;
	}
	public double getAmountSell() {
		return amountSell;
	}
	public void setAmountSell(double amountSell) {
		this.amountSell = amountSell;
	}
	public double getAmountBuy() {
		return amountBuy;
	}
	public void setAmountBuy(double amountBuy) {
		this.amountBuy = amountBuy;
	}
	public double getRate() {
		return rate;
	}
	public void setRate(double rate) {
		this.rate = rate;
	}
	public Date getTimePlaced() {
		return timePlaced;
	}
	public void setTimePlaced(Date timePlaced) {
		this.timePlaced = timePlaced;
	}
	public String getOriginatingCountry() {
		return originatingCountry;
	}
	public void setOriginatingCountry(String originatingCountry) {
		this.originatingCountry = originatingCountry;
	}
	public Date getTimeReceived() {
		return timeReceived;
	}
	public void setTimeReceived(Date timeReceived) {
		this.timeReceived = timeReceived;
	}
	public int getProcessingStatus() {
		return processingStatus;
	}
	public void setProcessingStatus(int processingStatus) {
		this.processingStatus = processingStatus;
	}
	public Date getTimeProcessed() {
		return timeProcessed;
	}
	public void setTimeProcessed(Date timeProcessed) {
		this.timeProcessed = timeProcessed;
	}
	public boolean isException() {
		return exception;
	}
	public void setException(boolean exception) {
		this.exception = exception;
	}
	public long getTradeId() {
		return tradeId;
	}
	public void setTradeId(long tradeId) {
		this.tradeId = tradeId;
	}
}

File: src/web_root/currencyfair/classes/Trade.php
<?php
class Trade {
	//Standard fields
	var $userId;
	var $currencyFrom;
	var $currencyTo;
	var $amountSell;
	var $amountBuy;
	var $rate;
	var $timePlaced;
	var $originatingCountry;
	
	//Meta-data
	var $timeReceived;
	var $processingStatus;
	var $timeProcessed;
	var $exception;

	function __construct() {
   		$this->processingStatus=0;
   		$this->exception = false;
	}
	
	//Perform basic input validation
	public function isValid() {
		return true;
	}

	public function populateFromArray($jsonArray) {
		if (array_key_exists('userId', $jsonArray)) {
			$this->userId = $jsonArray['userId'];
		}
		if (array_key_exists('currencyFrom', $jsonArray)) {
			$this->currencyFrom = $jsonArray['currencyFrom'];
		}
		if (array_key_exists('currencyTo', $jsonArray)) {
			$this->currencyTo = $jsonArray['currencyTo'];
		}
		if (array_key_exists('amountSell', $jsonArray)) {
			$this->amountSell = $jsonArray['amountSell'];
		}
		if (array_key_exists('amountBuy', $jsonArray)) {
			$this->amountBuy = $jsonArray['amountBuy'];
		}
		if (array_key_exists('rate', $jsonArray)) {
			$this->rate = $jsonArray['rate'];
		}
		if (array_key_exists('timePlaced', $jsonArray)) {
//			$this->timePlaced = new DateTime($jsonArray['timePlaced']);
			$this->timePlaced = $jsonArray['timePlaced'];
		}
		if (array_key_exists('currencyFrom', $jsonArray)) {
			$this->currencyFrom = $jsonArray['currencyFrom'];
		}
		if (array_key_exists('originatingCountry', $jsonArray)) {
			$this->originatingCountry = $jsonArray['originatingCountry'];
		}
	}
}
?>